ssl-config/src/main/java/com/palantir/remoting1/config/ssl/KeyStores.java

```diff
@@ -80,12 +83,43 @@ private KeyStores() {
      *         entry. The returned store will not have any password.
      */
     static KeyStore createTrustStoreFromCertificates(Path path) {
+        KeyStore keyStore;
+        try {
+            keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(null, null);
+        } catch (KeyStoreException | CertificateException | IOException | NoSuchAlgorithmException e) {
+            throw Throwables.propagate(e);
+        }
+
+        for (File currFile : getFilesForPath(path)) {
+            try(InputStream in = Files.newInputStream(currFile.toPath())) {
+                keyStore.setCertificateEntry(currFile.getName(), readX509Certificate(in));
+            } catch (IOException e) {
+                throw new RuntimeException(String.format(
+                        "IOException encountered when opening '%s'", currFile.toPath()), e);
+            } catch (CertificateException | KeyStoreException e) {
+                throw new RuntimeException(String.format(
+                        "Exception encountered when reading certificate in '%s'", currFile.toPath()), e);
+            }
+        }
+
+        return keyStore;
+    }
+
+    /**
+     * Returns a {@link KeyStore} created by loading the given certificates.
+     *
+     * @param certificatesByAlias
+     *        a map of X.509 certificate in PEM or DER format by the alias to load the certificate as.
+     */
+    static KeyStore createTrustStoreFromCertificates(Map<String, byte[]> certificatesByAlias) {
```

Comment:
I think `byte[]` is probably a weird way to carry the certificate bytes, maybe we should make an alias object so it's obvious what it is?